xbmc/windowing/wayland/ShellSurfaceWebOSShell.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ *  Copyright (C) 2017-2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ShellSurfaceWebOSShell.h"
+
+#include "CompileInfo.h"
+#include "Registry.h"
+#include "platform/Environment.h"
+#include "utils/log.h"
+
+using namespace KODI::WINDOWING::WAYLAND;
+using namespace std::placeholders;
+
+using namespace wayland;
+
+namespace
+{
+/*
+  WebOS always is 1920x1080
+*/
+constexpr int WEBOS_UI_WIDTH = 1920;
+constexpr int WEBOS_UI_HEIGHT = 1080;
+} // namespace
+
+CShellSurfaceWebOSShell::CShellSurfaceWebOSShell(IShellSurfaceHandler& handler,
+                                                 CConnection& connection,
+                                                 const wayland::surface_t& surface,
+                                                 const std::string& title,
+                                                 const std::string& class_)
+  : m_handler{handler}
+{
+  {
+    CRegistry registry{connection};
+    registry.RequestSingleton(m_shell, 1, 1);
+    registry.RequestSingleton(m_webos_shell, 1, 2);
+    registry.Bind();
+  }
+
+  m_shellSurface = m_shell.get_shell_surface(surface);
+
+  m_webos_shellSurface = m_webos_shell.get_shell_surface(surface);
+
+  m_webos_shellSurface.on_state_changed() = [this](std::uint32_t state) {
+    switch (state)
+    {
+      case (std::uint32_t)webos_shell_surface_state::fullscreen:
```

Comment:
this and also the others cases, do not use C casts
```suggestion
      case static_cast<uint32_t>(webos_shell_surface_state::fullscreen):
```